plugin/metric/metric.go

```diff
@@ -0,0 +1,67 @@
+package metric
+
+import (
+	"context"
+
+	"github.com/bytebase/bytebase/plugin/metric/collector"
+	"github.com/bytebase/bytebase/plugin/metric/reporter"
+	"go.uber.org/zap"
+)
+
+// Metric is the metric plugin.
+type Metric struct {
+	l         *zap.Logger
+	reporter  reporter.MetricReporter
+	executors map[string]collector.MetricCollector
+}
+
+// NewMetric returns a new instance of Controller
+func NewMetric(l *zap.Logger, key string, identifier string) *Metric {
+	return &Metric{
+		l:         l,
+		reporter:  reporter.NewSegmentReporter(l, key, identifier),
+		executors: make(map[string]collector.MetricCollector),
+	}
+}
+
+// Close will close the reporter client.
+func (m *Metric) Close() {
+	m.reporter.Close()
+}
+
+// Register will register a metric collector.
+func (m *Metric) Register(metricName reporter.MetricName, collector collector.MetricCollector) {
+	m.executors[string(metricName)] = collector
+}
+
+// CollectAndReport will exec all collectors and report metric.
+func (m *Metric) CollectAndReport(ctx context.Context) {
```

Comment:
The metric package should only do metric reporting, in other words, a common interface + what a segment client should do. In this case, the collector should not be part of it.